src/private/patient/js/yourVisits.js

```diff
@@ -0,0 +1,33 @@
+const getData = () => {
+    fetch(chooseProtocol() + window.location.host + "/api/patient/appointment")
+        .then(resp => resp.json())
+        .then(data => {
+            const tableBody = document.getElementById('visits-list-table')
+            console.log(data);
+            data.forEach(visit => {
+                console.log(visit);
+                row = document.createElement("tr");
+                row.setAttribute('id', `${visit.contact}`);
+
+                var date = new Date(visit.startDate * 1000);
+
+                row.innerHTML = `
+                    <td>${visit.doctor.firstName + " " + visit.doctor.lastName}</td>
+                    <td>${date.getUTCFullYear() +
+                        '-' + ('0' + date.getUTCMonth()).slice(-2) +
+                        '-' + ('0' + date.getUTCDate()).slice(-2)}</td>
+                    <td>${('0' + date.getUTCHours()).slice(-2) +
+                    ':' + ('0' + date.getUTCMinutes()).slice(-2)}</td>
+                `;
```

Comment:
Ciężko się to czyta jak tak pomieszane jest wyciągania daty z robieniem htmla.
Napisz funkcję, która jedyne co robi to na podstawie daty jako timestamp zwraca Ci datę gotową do wrzucenia i wyświetlenia, a w tym miejscu tylko użyj tej funkcji.
Szybkie googlowanie:
Masz złą wartość miesiąca, bo obiekt Date w javascript liczy miesiące od 0 do 11, więc przy konwersji trzeba dodać 1.
Tu masz snippet, który pokazuje jak działa zamiana, trochę prościej niż przez slice'owanie stringa.
let jsDate = new Date(timestamp * 1000);

```
jsDateValues = [
   jsDate.getFullYear(),
   jsDate.getMonth()+1,
   jsDate.getDate(),
   jsDate.getHours(),
   jsDate.getMinutes(),
   jsDate.getSeconds(),
];
alert(jsDateValues); //=> [2011, 3, 25, 23, 0, 0]
```

Reply:
Dziękuuuuuuje !!!